solr/core/src/test/org/apache/solr/handler/component/CustomSearchHandlerTest.java

```diff
@@ -0,0 +1,180 @@
+  private static class CustomResponseBuilder extends ResponseBuilder {
+
+    public CustomResponseBuilder(
+        SolrQueryRequest req, SolrQueryResponse rsp, List<SearchComponent> components) {
+      super(req, rsp, components);
+    }
+
+    @Override
+    protected ShardsInfoContainer newShardsInfoContainer() {
+      return new CustomShardsInfoContainer();
+    }
+  }
+
+  /**
+   * A container for recording shard to shard info mappings, retaining only the "shardAddress" part
+   * of the shard info.
+   *
+   * @see org.apache.solr.common.params.ShardParams#SHARDS_INFO
+   */
+  private static class CustomShardsInfoContainer extends ShardsInfoContainer {
+
+    private final List<Object> container = new ArrayList<>();
+
+    @Override
+    public void accept(String shardInfoName, NamedList<Object> shardInfoValue) {
+      final Object shardAddress = shardInfoValue.remove("shardAddress");
+      if (shardAddress != null) {
+        container.add(shardAddress);
+      }
+    }
+
+    @Override
+    public Object get() {
+      return container;
+    }
+  }
+
+  private static String COLLECTION;
+  private static int NUM_SHARDS;
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+
+    // decide collection name ...
+    COLLECTION = "collection" + (1 + random().nextInt(100));
+    // ... and shard/replica/node numbers
+    NUM_SHARDS = 1 + random().nextInt(3);
+    final int numReplicas = 2;
+    final int nodeCount = NUM_SHARDS * numReplicas;
+
+    // create and configure cluster
+    configureCluster(nodeCount).addConfig("conf", configset("cloud-dynamic")).configure();
+
+    // create an empty collection
+    CollectionAdminRequest.createCollection(COLLECTION, "conf", NUM_SHARDS, numReplicas)
+        .processAndWait(cluster.getSolrClient(), DEFAULT_TIMEOUT);
+    AbstractDistribZkTestBase.waitForRecoveriesToFinish(
+        COLLECTION, cluster.getSolrClient().getZkStateReader(), false, true, DEFAULT_TIMEOUT);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testShardsInfoContainerCustomisation() throws Exception {
+
+    // determine custom search handler name (the exact name should not matter)
+    final String customSearchHandlerName = "/custom_select" + random().nextInt();
+
+    // add custom handler
+    {
+      cluster
+          .getSolrClient()
+          .request(
+              new ConfigRequest(
+                  "{\n"
+                      + "  'add-requesthandler': {\n"
+                      + "    'name' : '"
+                      + customSearchHandlerName
+                      + "',\n"
+                      + "    'class' : '"
+                      + CustomSearchHandler.class.getName()
+                      + "',\n"
+                      + "    'components' : [ '"
+                      + QueryComponent.COMPONENT_NAME
+                      + "' ]\n"
+                      + "  }\n"
+                      + "}"),
```

Comment:
3/3 The layout here is not to my taste but it's what "spotless" made it to be.

Elsewhere we have https://github.com/apache/lucene-solr/blob/releases/lucene-solr/8.9.0/solr/core/src/test/org/apache/solr/handler/component/CustomHighlightComponentTest.java#L167-L174 and https://github.com/apache/lucene-solr/blob/releases/lucene-solr/8.9.0/solr/core/src/test/org/apache/solr/handler/component/CustomTermsComponentTest.java#L96-L103 layout.

But well for the three new files in this pull request it seemed logical to go with what "spotless" would (eventually) enforce for them.